samples/nrf9160/aws_iot/src/main.c

```diff
@@ -430,12 +456,8 @@ void main(void)
 #endif
 
 
-	date_time_update_async();
-
-	/** Sleep to ensure that time has been obtained before
-	 *  communication with AWS IoT.
-	 */
-	k_sleep(K_SECONDS(15));
+	date_time_update_async(date_time_event_handler);
+	k_sem_take(&date_time_obtained, K_FOREVER);
```

Comment:
Alternatively, set a timeout here, check the return code and print some sort of message if obtaining time failed within the timeout